x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/DeleteExpiredDataAction.java

```diff
@@ -54,15 +53,13 @@ private DeleteExpiredDataAction() {
 
         public static Request parseRequest(String jobId, XContentParser parser) {
             Request request = PARSER.apply(parser, null);
-            if (jobId != null) {
-                request.jobId = jobId;
-            }
+            request.jobId = jobId;
             return request;
         }
 
         private Float requestsPerSecond;
         private TimeValue timeout;
-        private String jobId = Metadata.ALL;
+        private String jobId;
```

Comment:
There are no `null` checks now on the object. Looking at the wire serialization, this would fail if jobId was `null` since it only calls `jobId = in.readString();` or `out.writeString(jobId);`